### PR TITLE
LibC: Implemented atof()

### PR DESCRIPTION
**Pull request by Quaker762**
`atof()` has now been implemented as part of the standard C library.
It supports scientific notation such as `1.2e-3` etc, ala the version
found as part of `glibc`.

It's a bit chunky, so there's probably room for optimisations here
and there, however, for now it works as intended (and allows Quake
to run).

### CODE REVIEW

File: Libraries/LibC/stdlib.cpp [hunk 1]
```diff
@@ -338,10 +401,10 @@ char* mkdtemp(char* pattern)
         struct stat st;
         int rc = lstat(pattern, &st);
         if (rc < 0 && errno == ENOENT) {
-	    if (mkdir(pattern, 0700) < 0)
+            if (mkdir(pattern, 0700) < 0)
                 return nullptr;
-	    return pattern;
-	}
+            return pattern;
+        }
```
Comment (Quaker762) [inline Libraries/LibC/stdlib.cpp]: @awesomekling side effect of `clang-format` sorry

Comment (awesomekling) [reply Libraries/LibC/stdlib.cpp]: np lol

File: Libraries/LibC/stdlib.cpp [hunk 2]
```diff
@@ -196,8 +194,73 @@ float strtof(const char* str, char** endptr)
...
 double atof(const char* str)
 {
-    dbgprintf("LibC: atof: '%s'\n", str);
-    ASSERT_NOT_REACHED();
+    size_t len = strlen(str);
+    size_t weight = 1;
+    int expVal = 0;
+    double value = 0.0f;
+    double fraction = 0.0f;
+    bool hasSign = false;
```
Comment (awesomekling) [inline Libraries/LibC/stdlib.cpp]: Please use snake_case for variables.

File: Libraries/LibC/stdlib.cpp [hunk 3]
```diff
@@ -196,8 +194,73 @@ float strtof(const char* str, char** endptr)
...
+        }
+
+        if (str[i] < '0' || str[i] > '9' || expVal != 0)
+            continue;
+
+        if (isFractional) {
+            fraction *= 10;
+            fraction += str[i] - '0';
+            weight *= 10;
+            ;
```
Comment (awesomekling) [inline Libraries/LibC/stdlib.cpp]: Unnecessary semicolon by itself here.

**Comment by awesomekling** [comment]
Awesome, nice work Jesse :)

**Pull request merged**